### PR TITLE
feat: new search boilerplate with feature flag enabled. 

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Feat

This PR solves [FX-4651]

This PR resolves the followings :

- Add fx-force-search-dropdown-improvements feature flag in [Unleash](https://unleash.artsy.net/features)
- Add a new search dropdown component that will not display anything when feature flag is enabled.
- Add `AutocompleteInput` from palette as a new Search input container

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4651]: https://artsyproduct.atlassian.net/browse/FX-4651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -82,6 +84,10 @@ export const NavBar: React.FC = track(
   const isLoggedIn = Boolean(user)
   const showNotificationCount = isLoggedIn && !showMobileMenu
 
+  const isSearchDropDownImprovementsEnabled = useFeatureFlag(
+    "fx-force-search-dropdown-improvements"
+  )
+
```
Comment (rajsam003) [inline src/Components/NavBar/NavBar.tsx:90]: I am not sure why it's not getting enabled. Any idea, please?

Comment (dimatretyak) [reply src/Components/NavBar/NavBar.tsx:90]: It's enabled for me. I also updated the strategies and the list of user ids (enabled ff only for you and me) in Unleash. Could you please check?

for-sam

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const NewSeacrhBar = () => {
+  return (
+    <ClassI18n>
+      {({ t }) => (
+        <Form
+          action="/search"
+          method="GET"
+          onSubmit={() => console.log("Pressed ")}
+        >
+          <Text> I am the new Search bar</Text>
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:14]: From "Acceptance Criteria" section for [FX-4651](https://artsyproduct.atlassian.net/browse/FX-4651)

> And user typed 2 or more characters in the search input
> User sees a new search dropdown component (which is empty for now)

if it's not difficult, can you please add this too as part of the current PR? or did you want to add this as part of the next PR?

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Text } from "@artsy/palette"
+import { ClassI18n } from "System/i18n/ClassI18n"
+import styled from "styled-components"
+
+export const NewSeacrhBar = () => {
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:5]: small typo in the name: `NewSeacrhBar` -> `NewSearchBar`

**Review by dimatretyak — DISMISSED** [review]
LGTM 👍

File: src/Components/NavBar/NavBar.tsx (hunk 1 above)
Comment (rajsam003) [reply src/Components/NavBar/NavBar.tsx:90]: Its still the same 🤔

image

File: src/Components/Search/NewSearch/NewSearchBar.tsx (hunk 2 above)
Comment (rajsam003) [reply src/Components/Search/NewSearch/NewSearchBar.tsx:14]: Ya sure. I am updating the PR

Comment (dimatretyak) [reply src/Components/Search/NewSearch/NewSearchBar.tsx:14]: thanks!

File: src/Components/NavBar/NavBar.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/Components/NavBar/NavBar.tsx:90]: > [Force] Warning: cannot find flagName in featureFlags: 
do you see this warning in your console?

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        spellCheck={false}
+        options={[]}
+        onChange={handleChange}
+      />
+    )
+  }
+
+  return (
+    <Box flex={1}>
+      <ClassI18n>
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:23]: **suggestion**: use `t` from `useTranslation` hook (like [here](https://github.com/artsy/force/blob/9faf1c2185c05545bef2d4ea8a086cd648d0a568/src/Apps/Artist/Components/ArtistHeader/ArtistHeader.tsx#L27))

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const renderAutosuggestComponent = (t, { xs }) => {
+    return (
+      <AutocompleteInput
+        placeholder={xs ? t`navbar.searchArtsy` : t`navbar.searchBy`}
+        spellCheck={false}
+        options={[]}
+        onChange={handleChange}
+      />
+    )
+  }
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:19]: **suggestion**: separate this into a new component

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <Box flex={1}>
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:22]: i think it's better to use `Flex` component

File: src/Components/Search/NewSearch/NewSearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { AutocompleteInput, Box } from "@artsy/palette"
+import { ClassI18n } from "System/i18n/ClassI18n"
+import { Media } from "Utils/Responsive"
+
+export const NewSearchBar = () => {
+  const handleChange = ({ target: { value } }) => {
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBar.tsx:6]: **suggestion**: please use `TS` defs

**Review by dimatretyak — APPROVED** [review]
you can merge the PR when [these changes](https://github.com/artsy/force/pull/12268/files#r1171318945) are added.

File: src/Components/Search/NewSearch/NewSearchBar.tsx (hunk 6 above)
Comment (rajsam003) [reply src/Components/Search/NewSearch/NewSearchBar.tsx:7]: With Flex we need more props. Box is better in my opinion.

**Pull request merged**